content/articles/javafx-editable-table-view/index.md

```diff
@@ -0,0 +1,442 @@
+
+JavaFX, being focused on UI, has ways to represent data in tabular format. The problem is that one can only view but not edit each cell individually since they aren't editable. This article has solutions to this issue. By the end of the article, the reader will be able to create an editable Table View. That will enable easy data updates directly on the Table being displayed.
+
+### Table of Contents
+
+- [Key takeaways](#key-takeaways)
+- [Pre-requisites](#pre-requisites)
+- [Article structure](#article-structure)
+- [Project initialization](#project-initialization)
+  - [Initial Project Structure](#initial-project-structure)
+  - [New Project Structure](#new-project-structure)
+- [Set up the project, tools, and development environment](#set-up-the-project-tools-and-development-environment)
+- [Table creation and design](#table-creation-and-design)
+- [Add data items on the TableView](#add-data-items-on-the-tableview)
+  - [Define Users model](#define-users-model)
+  - [Setup the controller](#setup-the-controller)
+- [Make the Table cells editable](#make-the-table-cells-editable)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Key takeaways
+
+At the article's tip, the reader will be able to do the following:
+
+- Create a TableView
+- Format the Table
+- Add data items on the TableView
+- Make the TableView editable through the use of its controller
+- Get the edited items
+
+### Pre-requisites
+
+This is an intermediate-level Java user article. For one to easily follow up with the article, the following are the
+needed skills and tools:
+
+- A good Java IDE. The recommendation is the use of the latest IntelliJ Ultimate version.
+- A good internet connection
+- Java SDK installed on the machine. JDK 17 was used for the screenshots.
+
+> **NOTE**: The system requirements and specifications may change over time. Make sure to be watching out for new ones and how to implement them in the project.
+
+### Article structure
+
+In the article, the following steps will be followed to accomplish the points noted in
+the [Key TakeAways](#key-takeaways) section:
+
+- Project initialization
+- Table creation and design
+- Adding dummy data to the table
+- Make the Table data editable
+- Fetch the contents edited
+
+Let's begin with the first step:
+
+### Project initialization
+
+To set up a new JavaFX project, do the following:
+
+- Open the IDE and click on the '**New Project**' button
+- Select JavaFX as the Framework
+- Set the following in the window:
+    - **Name**: editable-tableview
+    - **Group**: com.table
+    - **Artifact**: editabletableview
+
+This is shown below:
+
+![New JavaFX editable tableview project](new-javafx-editable-tableview-project.png "New JavaFX editable tableview project")
```

Comment:
Please edit the image path.

It would be something like

`/engineering-education/[ARTICLE-DIRECTORY]/[IMAGE-NAME-WITH-EXTENSION]`

Reply:
Do this for all the images

Reply:
Ok, got you

Reply:
Done!